### PR TITLE
[Feature] add reid dataset

### PR DESCRIPTION
**Pull request by ToumaKazusa3**
Hi~
This PR add the following features:
1. Add `SeqReIDFormatBundle` pipeline. It mainly formats the gt_label field, which is missing in `SeqDefaultFormatBundle`.
2. Add ReIDDataset
3. Add the unitest of the above.

### CODE REVIEW

File: mmtrack/datasets/pipelines/formatting.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import numpy as np
+from mmcls.datasets.builder import PIPELINES as CLS_PIPELINES
 from mmcv.parallel import DataContainer as DC
-from mmdet.datasets.builder import PIPELINES
+from mmdet.datasets.builder import PIPELINES as DET_PIPELINES
```
Comment (OceanPang) [inline mmtrack/datasets/pipelines/formatting.py:4]: We should only use a `PIPELINE`, it doesn't matter whether it is for cls/det/tracking

File: mmtrack/datasets/reid_dataset.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+import torch
+from mmcls.datasets import DATASETS, BaseDataset
+
+
+@DATASETS.register_module()
+class ReIDDataset(BaseDataset):
+    """Dataset for ReID Dataset.
+
+    Args:
+        load_as_video (bool): If True, using COCOVID class
```
Comment (OceanPang) [inline mmtrack/datasets/reid_dataset.py:14]: `load_as_video` seems useless here. We should remove it from this function. If there are some conflicts with former build functions, adapt the building function

Comment (ToumaKazusa3) [reply mmtrack/datasets/reid_dataset.py:14]: `load_as_video` is used in `eval_hooks`.
https://github.com/open-mmlab/mmtracking/blob/master/mmtrack/core/evaluation/eval_hooks.py#L14

File: mmtrack/datasets/pipelines/formatting.py [hunk 3]
```diff
@@ -350,3 +350,33 @@ def __call__(self, results):
...
+
+
+@PIPELINES.register_module()
+class SeqReIDFormatBundle(SeqDefaultFormatBundle):
+    """Sequence ReID formatting bundle.
+
+    It simplifies the pipeline of formatting common fields, including "img",
+    "img_metas" and "gt_label". These fields are formatted as follows.
+
+    - img: (1)transpose, (2)to tensor, (3)to DataContainer (stack=True)
```
Comment (OceanPang) [inline mmtrack/datasets/pipelines/formatting.py:362]: add a blank after the bracket, same below

File: mmtrack/datasets/reid_dataset.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+    def __init__(self,
+                 pipeline,
+                 triplet_sampler=dict(num_ids=1, ins_per_id=1),
+                 *args,
+                 **kwargs):
+        super().__init__(pipeline=[], *args, **kwargs)
+        self.triplet_sampler = triplet_sampler
+        # for DistributedGroupSampler and GroupSampler
+        self.flag = np.zeros(len(self), dtype=np.uint8)
```
Comment (OceanPang) [inline mmtrack/datasets/reid_dataset.py:29]: The `flag` is not used? Remove it

**Review by OceanPang — APPROVED** [review]

**Pull request merged**